integration_tests/test_rough_align.py

```diff
@@ -534,192 +464,40 @@ def test_apply_rough_alignment_transform(render, montage_stack, test_do_rough_al
-    ex = {
-        "render": render_params,
-        "input_stack": one_tile_montage,
-        "image_directory": image_directory,
-        "imgformat": "png",
-        "scale": 0.1,
-        "minZ": -1,
-        "maxZ": -1,
-    }
-    ex['output_json'] = os.path.join(image_directory, 'output.json')
-
-    ex2 = dict(ex, **{'minZ': 1021, 'maxZ': 1020})
-    ex3 = dict(ex2, **{'minZ': 1 , 'maxZ': 2})
-
-    #stack_has_mipmaps = check_stack_for_mipmaps(render, ex['input_stack'], [1020])
-    # tempjson = create_tilespecs_without_mipmaps(render, ex['input_stack'], 0, 1020)
-
-    # generate tilespecs used for rendering stack input
-    maxlvl = 1
-    tspecs = create_tilespecs_without_mipmaps(render, ex['input_stack'], maxlvl, 1020)
-    ts_levels = {int(i) for l in (ts.ip.levels for ts in tspecs.tilespecs) for i in l}
-    #ts_levels = {int(i) for l in (ts.ip.levels for ts in tspecs) for i in l}
-    # levels > maxlevel should not be included if
-    assert not ts_levels.difference(set(range(maxlvl + 1)))
-
-    mod = RenderSectionAtScale(input_data=ex, args=[])
-    mod.run()
-
-    with open(ex['output_json'], 'r') as f:
-        js = json.load(f)
-
-    out_dir = os.path.join(image_directory, render_params['project'], js['temp_stack'], 'sections_at_0.1/001/0')
-    assert(os.path.exists(out_dir))
-
-    files = glob.glob(os.path.join(out_dir, '*.png'))
-    for fil in files:
-        img = os.path.join(out_dir, fil)
-        assert(os.path.exists(img) and os.path.isfile(img) and os.path.getsize(img) > 0)
-
-    mod = RenderSectionAtScale(input_data=ex2, args=[])
-    with pytest.raises(RenderModuleException):
-        mod.run()
-
-    mod = RenderSectionAtScale(input_data=ex3, args=[])
-    with pytest.raises(RenderModuleException):
-        mod.run()
-
-    ex2 = dict(ex1, **{
-        'set_new_z': False,
-        'output_stack': 'failed_output',
-        'render': dict(ex1['render'], **render_params),
-        'montage_stack': montage_stack,
-        'lowres_stack': test_do_rough_alignment,
-        'prealigned_stack': None,
-        'tilespec_directory': str(tmpdir_factory.mktemp('scratch')),
-        'old_z': [1020, 1021, 1022],
-        'scale': 0.1,
-        'pool_size': pool_size,
-        'output_json': str(tmpdir_factory.mktemp('output').join('output.json')),
-        'loglevel': 'DEBUG'
-        })
-    with pytest.raises(RenderModuleException):
-        renderapi.stack.set_stack_state(ex2['lowres_stack'],'LOADING',render=render)
-        renderapi.stack.delete_section(ex2['lowres_stack'],1021,render=render)
-        renderapi.stack.set_stack_state(ex2['lowres_stack'],'COMPLETE',render=render)
-        mod = ApplyRoughAlignmentTransform(input_data=ex2, args=[])
-        mod.run()
-        zvalues = renderapi.stack.get_z_values_for_stack(ex2['output_stack'],render=render)
-        assert(1021 not in zvalues)
-
-
-def make_montage_stack_without_downsamples(render, montage_stack, tmpdir_factory):
```

Comment:
This functionality should be dropped from the modules before it is left untested.  The complexity of the codebase is such that code coverage can be okay even if major things are untested.

Reply:
fucntionality added again